nixos/modules/services/mail/maddy.nix

```
@@ -215,6 +214,65 @@ in {
         '';
       };
 
+      tls = {
+        loader = mkOption {
+          type = with types; nullOr (enum [ "file" "off" ]);
+          default = "off";
+          description = lib.mdDoc ''
+            TLS certificates are obtained by modules called "certificate
+            loaders". Currently only the file loader is supported which reads
+            certificates from files specifying the options `keyPaths` and
+            `certPaths`.
+          '';
+        };
+
+        certificates = mkOption {
+          type = with types; nullOr (listOf (submodule {
```

Why `null` instead of an empty list? I think an empty list is the better choice for "no certificates", I'd see `null` as a "certificates not applicable", but that's handled via `tls = null;`.

```suggestion
          type = with types; (listOf (submodule {
```